services/chrome-web-store/chrome-web-store-base.js

```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+const { BaseService } = require('..')
+const chromeWebStore = require('chrome-web-store-item-property')
+const { checkErrorResponse } = require('../../lib/error-helper')
+
+module.exports = class BaseChromeWebStoreService extends BaseService {
+  async fetch({ storeId }) {
+    try {
+      return await chromeWebStore(storeId)
+    } catch (e) {
+      return checkErrorResponse.asPromise({})({ buffer: '', res: e })
```

Comment:
Other than here we only use `checkErrorResponse.asPromise` in one place and I was thinking we could inline it. This is actually using it to re-render a chrome web store error object, yea? It seems a little opaque though fine for now.

Could you keep the `networkOff()` connection test since effectively we're not using `BaseService._request`?

Reply:
Yep. The `HTTPError` from `chrome-web-store-item-property` has a `statusCode` property, which is the only thing we're looking at inside `checkErrorResponse` so it seemed reasonable to use that helper here given the object "quacks like a duck", as it were. I've added a comment here to explain it.

Reply:
Just looking at the `networkOff()` case, but I'm now really confused about where we usually catch this. How is this case normally handled?

Reply:
`BaseService._request` delegates to `checkErrorResponse.asPromise` which emits `Inaccessible` for 5xx errors. Is that what you're asking? 😀

Reply:
Ah no. In the 5XX error case, that's fine. We can pass the exception object to `checkErrorResponse` because we've got a `statusCode` but in the case where we just get no response at all  (simulated with `networkOff()` in the tests), there is a different error which doesn't have a `statusCode` because we didn't get a response. That's the bit of handling I'm struggling to track down.

Reply:
`.networkOff()` causes a request to immediately trigger an ENETUNREACH error.

I believe that gets passed as the `err` parameter to the `request` callback – that callback gets `(error, response, body)` – which eventually is re-emitted here:

https://github.com/badges/shields/blob/0941b6c39ca0021378abfe825b34f428ef553d86/core/base-service/legacy-request-handler.js#L245

This is the code in nock that generates it; as you say there is no status code:

https://github.com/nock/nock/blob/8493b80656a1b4c6952fa74ba814f1ee56e9404a/lib/intercept.js#L32-L34

Reply:
I was looking for <pre><strong>throw</strong> new Inaccessible</pre> which is why I wasn't matching this. Thought I was going out of my mind there.

Reply:
More reasons to get rid of legacy code!